src/NuGet.Services.AzureSearch/Db2AzureSearch/Db2AzureSearchCommand.cs

```diff
@@ -0,0 +1,228 @@
+                    }
+
+                    foreach (var action in indexActions.Hijack)
+                    {
+                        hijackActions.Enqueue(action);
+                    }
+
+                    var searchBatchesTask = PushFullBatchesAsync(_searchIndexClient, searchActions);
+                    var hijackBatchesTask = PushFullBatchesAsync(_hijackIndexClient, hijackActions);
+                    await Task.WhenAll(searchBatchesTask, hijackBatchesTask);
+                }
+
+                var searchFinishTask = IndexAsync(_searchIndexClient, searchActions);
+                var hijackFinishTask = IndexAsync(_hijackIndexClient, hijackActions);
+                await Task.WhenAll(searchFinishTask, hijackFinishTask);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(
+                    0,
+                    ex,
+                    "An exception was thrown while processing package ID {PackageId}.",
+                    work?.PackageId ?? "(last batch...)");
+                throw;
+            }
+        }
+
+        private async Task PushFullBatchesAsync(
+            ISearchIndexClientWrapper indexClient,
+            Queue<IndexAction<KeyedDocument>> actions)
+        {
+            while (actions.Count > _options.Value.AzureSearchBatchSize)
+            {
+                var batch = new List<IndexAction<KeyedDocument>>();
+                while (batch.Count < _options.Value.AzureSearchBatchSize)
+                {
+                    batch.Add(actions.Dequeue());
+                }
+
+                await IndexAsync(indexClient, batch);
+            }
+        }
+
+        private async Task IndexAsync(
+            ISearchIndexClientWrapper indexClient,
+            IReadOnlyCollection<IndexAction<KeyedDocument>> batch)
+        {
+            if (batch.Count == 0)
+            {
+                return;
+            }
+
+            await Task.Yield();
+
+            _logger.LogInformation(
+                "Pushing batch of {BatchSize} to index {IndexName}.",
+                batch.Count,
+                indexClient.IndexName);
+            var batchResults = await indexClient.Documents.IndexAsync(new IndexBatch<KeyedDocument>(batch));
+            const int errorsToLog = 5;
+            var errorCount = 0;
+            foreach (var result in batchResults.Results)
+            {
+                if (!result.Succeeded)
+                {
+                    if (errorCount < errorsToLog)
+                    {
+                        _logger.LogError(
+                            "Indexing document with key {Key} failed. {StatusCode}: {ErrorMessage}",
+                            result.Key,
+                            result.StatusCode,
+                            result.ErrorMessage);
+                    }
+
+                    errorCount++;
+                }
+            }
+
+            if (errorCount > errorsToLog)
```

Comment:
>errorCount > errorsToLo [](start = 16, length = 23)

Shouldn't this be `errorCount > 0`? Isn't there pending actions that need to be indexed? If this is intended behavior, please add a comment explaining this :)

Reply:
Oh yeah, good point.